addons/hr_timesheet/tests/test_timesheet.py

```diff
@@ -806,3 +806,35 @@ def create_project_update():
         create_timesheet(self.task2, 10)
         project_update_vals_list = create_project_update()
         self.assertListEqual(project_update_vals_list, [12, 15, 125])
+
+    def test_sub_task_remaining_hours(self):
+        """
+        Check whether remaining hours are correctly computed for subtasks.
+            Steps:
+            - Create subtask
+            - Set 20 allocated hours on project
+            - Set 0 allocated hours on parent task
+            - Create a timesheet for 10 hours on the subtask
+            - Check remaining hours (-10)
+            - Set 5 allocated hours on parent
+            - Check remaining hours
+        """
+        subtask = self.env['project.task'].create({
+            'name': 'Sub Task for Remaining Hours Test',
+            'project_id': self.project_customer.id,
+            'parent_id': self.task1.id,
+        })
+
+        self.project_customer.allocated_hours = 20.0
+        self.task1.allocated_hours = 0
+
```

Comment:
I'd create a timesheet for the parent task too, and check that the remaining hours on the parent task is correctly computed into negative values too (since the project has allocated hours > 0)

Reply:
> I'd create a timesheet for the parent task too, and check that the remaining hours on the parent task is correctly computed into negative values too (since the project has allocated hours > 0)

According to the task specification, I had only added test cases for the sub-tasks. Now, I have checked the remaining hours for both the parent task and the sub-tasks.